src/client/components/MealById.js

```diff
@@ -0,0 +1,158 @@
+import React, { useContext, useState } from "react";
+import Header from "./Header";
+import Footer from "./Footer";
+import { useParams } from "react-router-dom";
+import DataContext from "./DataContext";
+import { Link } from "react-router-dom";
+
+export default function MealById() {
+  const { newMeals, reviews } = useContext(DataContext);
+  const { id } = useParams();
+  const [inputState, setInputstate] = useState(false);
+  const [name, setName] = useState("");
+  const [phoneNumber, setPhoneNumber] = useState("");
+  const [numberOfGuests, setNumberOfGuests] = useState("");
+  const [email, setEmail] = useState("");
+
+  function newReservation() {
+    {
+      if (email !== "") {
+        setInputstate(true);
+        fetch("/api/reservations", {
+          method: "POST",
+          headers: { "Content-Type": "application/json" },
+          body: JSON.stringify({
+            contact_name: name,
+            contact_phonenumber: phoneNumber,
+            contact_email: email,
+            meal_id: id,
+            number_of_guests: numberOfGuests,
+          }),
+        })
+          .catch((e) => {
+            setError(e);
+            alert("please enter the Valid details");
+          })
+          .finally(() => {
+            setInputstate(false);
+            alert("Your Reservation has been saved. Enjoy the Experience.");
+          });
```

Comment:
This pattern is also repeated in every page component, maybe it would make sense to extract a function or a hook that can handle this?